### DISCUSSION TITLE
How to control the main loop ?

### DISCUSSION THREAD

**Discussion by cgimenez** [hajimehoshi/ebiten#3167] [Q&A] [answered]
Hi all !

I would like to use ebiten as a rendering engine, the main loop being controled elsewhere
For now, I'm using Raylib bindings this way (drawing to offscreen texture, then when needed call begindraw/enddraw), but I need to switch to the only "wasm compatible" solution I'm aware of : ebiten
I searched through all issues, discussions but didn't found an example of how to do

**Comment by hajimehoshi** [comment, marked as answer]
Ebitengine doesn't provide a way to use an external main loop (except for mobiles).

**Reply by cgimenez** [reply]
Thanks for your answer, I'll search for another solution...
